src/lsstseries/ensemble.py

```diff
@@ -348,12 +351,26 @@ def batch(self, func, *args, meta=None, use_map=True, compute=True, **kwargs):
         if meta is None:
             meta = (self._id_col, type(self._id_col))  # return a series of ids
 
+        if on is None:
+            on = self._id_col  # Default grouping is by id_col
+
+        # Handle object columns to group on
+        source_cols = list(self._source.columns) + [self._source.index.name]
+        object_cols = list(self._object.columns) + [self._object.index.name]
+        object_group_cols = [col for col in on if (col in object_cols) and (col not in source_cols)]
```

Comment:
I ran patched in this branch and ran some tests and it looks like it is dropping the index. I *think* you could get the same results with:

```
        source_cols = list(self._source.columns)
        object_cols = list(self._object.columns)
        object_group_cols = [col for col in on if (col in object_cols) and (col not in source_cols)]
```

when the indices match in both tables.

The groupby on index still works, because lines 362-366 preserve the index.

So this all looks good to me.